### PR TITLE
[Gemini] add unitests to check gemini correctness

### PR DESCRIPTION
**Pull request by feifeibear**
Test gemini correctness on 1 process using various models. Currently, gpt and bert are correct. However, some special models still failed, for example resnet, and inline_op_model. I will fix these bugs ASAP.

### CODE REVIEW

File: tests/components_to_test/gpt.py [hunk 1]
```diff
@@ -15,8 +17,7 @@ def generate(self):
         input_ids = torch.randint(0,
                                   DummyDataLoader.vocab_size, (DummyDataLoader.batch_size, DummyDataLoader.seq_len),
                                   device=get_current_device())
-        attention_mask = torch.ones_like(input_ids)
-        return input_ids, attention_mask
```
Comment (ver217) [inline tests/components_to_test/gpt.py:19]: Why not generate mask in data loader?

Comment (feifeibear) [reply tests/components_to_test/gpt.py:19]: 因为就这个GPT特殊，他generate两个输入，其他都一个。我的run

Comment (feifeibear) [reply tests/components_to_test/gpt.py:19]: 这个gpt目前基本上没有在单测里使用上。
